Src/IronPython/Runtime/Binding/PythonProtocol.Operations.cs

```diff
@@ -857,6 +857,11 @@ internal static partial class PythonProtocol {
             }
         }
 
+        private static void MakeNegatedCompareReturn(ConditionalBuilder/*!*/ bodyBuilder, Expression retCondition, Expression/*!*/ retValue, bool isReverse, Type retType) {
+            Expression newRetVal = Ast.Not(Ast.Call(Compiler.Ast.AstMethods.IsTrue, retValue));
```

Comment:
I think that the dynamic conversion is necessary here because we have methods that return an object type, which can be a boolean or NotImplemented.